texthook/engine/engine.cc

```diff
@@ -2141,6 +2141,48 @@ bool InsertBGI3Hook()
   return true;
 }
 #endif // 0
+
+bool InsertBGI4Hook()
+{
+    /*
+    * Sample games:
+    * https://vndb.org/v26664
+    */
+    bool found = false;
+    const BYTE pattern[] = {
+        0x55,                           // 55               push ebp
+        0x8b,0xec,                      // 8BEC             mov ebp,esp
+        0x53,                           // 53               push ebx
+        0x56,                           // 56               push esi
+        0x57,                           // 57               push edi
+        0x33, 0xFF,                     // 33 FF            xor edi,edi
+        0xE8, 0x23, 0xFD, 0xFF, 0xFF,   // E8 23FDFFFF      call saclet.exe+A0990
```

Comment:
Seems unlikely that the function call bytes would be the same for other games, can't XX4 be used here?